packages/core/src/use-params.ts

```diff
@@ -98,6 +98,10 @@ export function useParam(
   const params = useParams()
   const rawValue = params[key]
 
+  if (typeof rawValue === "undefined") {
+    return undefined
+  }
+
   if (returnType === "number") {
     // Special case because Number("") === 0
     if (rawValue === "") {
```

Comment:
Right after this line, it is returning `NAN`. should that be returning `undefined` instead?

Reply:
I'm not sure how to view the entire file and still provide comments on unchanged code :(

Reply:
I wish it did returned `NaN` too -- I have a WIP where I refactor the whole thing and suggest to never return `NaN` but it would be a separate change imo

Reply:
@katt I think we should go ahead and do the NaN change with this

Reply:
Done this now - also see #1491